packages/parse5-parser-stream/lib/index.ts

```diff
@@ -0,0 +1,105 @@
+import { Writable } from 'node:stream';
+import { Parser, ParserOptions } from 'parse5/dist/parser/index.js';
+import type { TreeAdapterTypeMap } from 'parse5/dist/tree-adapters/interface.js';
+import type { DefaultTreeAdapterMap } from 'parse5/dist/tree-adapters/default.js';
+
+/**
+ * Streaming HTML parser with scripting support.
+ * A [writable stream](https://nodejs.org/api/stream.html#stream_class_stream_writable).
+ *
+ * @example
+ *
+ * ```js
+ * const ParserStream = require('parse5-parser-stream');
+ * const http = require('http');
+ *
+ * // Fetch the page content and obtain it's <head> node
+ * http.get('http://inikulin.github.io/parse5/', res => {
+ *     const parser = new ParserStream();
+ *
+ *     parser.once('finish', () => {
+ *         console.log(parser.document.childNodes[1].childNodes[0].tagName); //> 'head'
+ *     });
+ *
+ *     res.pipe(parser);
+ * });
+ * ```
+ *
+ */
+export class ParserStream<T extends TreeAdapterTypeMap = DefaultTreeAdapterMap> extends Writable {
+    private lastChunkWritten = false;
+    private writeCallback: null | (() => void) = null;
+    private pausedByScript = false;
+
+    public parser: Parser<T>;
+    private pendingHtmlInsertions: string[] = [];
+    /** The resulting document node. */
+    public document: T['document'];
+
+    constructor(options?: ParserOptions<T>) {
```

Comment:
another one for future: require accessibility modifiers on all methods/properties (will open an issue for these)

Reply:
opened #376